### PR TITLE
Contribution Guidelines

### LINKED ISSUE #368: Contribution Guidelines
Follow to the discussion on the [forum](https://forum.clr.fund/t/who-should-decide-what-prs-to-merge-and-how-to-resolve-disputes). 
The idea is to create a transparent guideline for making contributions and outline the process to creating an issue, request a code review, and have a PR merged into one of the repos that we can include in each project.

The working document can be found here: https://hackmd.io/h4BoM8amR4e2X-RW-HC2sA and will remain open to the public for comment or contribution over the next week.

- [x] Wait 7 days
- [x] Finalize CONTRIBUTING.MD
- [x] Create feature branch
- [x] PR into /develop

### PR DESCRIPTION
**Pull request by daodesigner**
First stab at contributor guidelines [CONTRIBUTING.md ](https://hackmd.io/h4BoM8amR4e2X-RW-HC2sA)
It’s open edit at the moment, so anyone is free to chime in (be civil please). Will leave it up for the next few days.

closes #368 

- [x]  Wait 7 days
- [x] Finalize CONTRIBUTING.MD

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+- If you opened an issue, but figured out the answer later on your own, comment on the issue to let people know, then close the issue. Even documenting that outcome is a contribution to the project.
+
+## Opening a pull request
+You should usually open a pull request in the following situations:
+
+- Submit trivial fixes (for example, a typo, a broken link or an obvious error)
+- Start work on a contribution that was already asked for, or that you’ve already discussed, in an issue
+- A pull request doesn’t have to represent finished work. It’s usually better to open a pull request early on, so others can watch or give feedback on your progress. Just mark it as a “WIP” (Work in Progress) in the subject line. You can always add more commits later.
+
+If the project is on GitHub, here’s how to submit a pull request:
```
Comment (xuhcc) [inline CONTRIBUTING.md]: clrfund is on github, so this part can be omitted

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+😇 “I’m disappointed you can’t support my use case, but as you’ve explained it only affects a minor portion of users, I understand why. Thanks for listening.”
+
+😢 “Why won’t you support my use case? This is unacceptable!”
+
+### Above all, keep it classy. 
+Open source is made up of collaborators from all over the world. Context gets lost across languages, cultures, geographies, and time zones. In addition, written communication makes it harder to convey a tone or mood. Assume good intentions in these conversations. It’s fine to politely push back on an idea, ask for more context, or further clarify your position. Just try to leave the internet a better place than when you found it.
+
+## Gathering context
+Before doing anything, do a quick check to make sure your idea hasn’t been discussed elsewhere. Skim the project’s README, issues (open and closed), Discourse and Discord. You don’t have to spend hours going through everything, but a quick search for a few key terms goes a long way.
```
Comment (xuhcc) [inline CONTRIBUTING.md]: Is there an official clrfund discord channel? I only know about public telegram group.

Also, I suggest creating a link here.

Comment (daodesigner) [reply CONTRIBUTING.md]: @auryn-macmillan ok to put invite links?

File: CONTRIBUTING.md (hunk 1 above)
Comment (daodesigner) [reply CONTRIBUTING.md]:
```
- If the project is on GitHub, here’s how to submit a pull request:
+ Here’s how to submit a pull request:
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (samajammin) [reply CONTRIBUTING.md]: Came here to suggest the same 😄 - we should add links, as the Discord & Discourse aren't easily findable.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+- If you see an open issue that you want to tackle, comment on the issue to let people know you’re on it. That way, people are less likely to duplicate your work.
+- If an issue was opened a while ago, it’s possible that it’s being addressed somewhere else, or has already been resolved, so comment to ask for confirmation before starting work.
+- If you opened an issue, but figured out the answer later on your own, comment on the issue to let people know, then close the issue. Even documenting that outcome is a contribution to the project.
+
+## Opening a pull request
+You should usually open a pull request in the following situations:
+
+- Submit trivial fixes (for example, a typo, a broken link or an obvious error)
+- Start work on a contribution that was already asked for, or that you’ve already discussed, in an issue
+- A pull request doesn’t have to represent finished work. It’s usually better to open a pull request early on, so others can watch or give feedback on your progress. Just mark it as a “WIP” (Work in Progress) in the subject line. You can always add more commits later.
```
Comment (samajammin) [inline CONTRIBUTING.md:80]: You can also make PRs as "drafts": https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request

**Comment by samajammin** [comment]
Looks great to me! Thanks for creating this 😍

**Review by wackerow — COMMENTED** [review]
This looks great! Thanks @daodesigner!

Agree with the other comments... this looks great, but would suggest adding the discord invite link, and perhaps also the link about how to properly link your PR to issues (https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

Thanks for putting this together! 💯

**Comment by auryn-macmillan** [comment]
I think we're probably good to go ahead and merge this, yeah? Any objections?

**Review by auryn-macmillan — APPROVED** [review]
LGTM

**Pull request merged**